x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/users/all/query.all_users.dsl.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import type { ISearchRequestParams } from '../../../../../../../../../src/plugins/data/common';
+import { Direction } from '../../../../../../common/search_strategy';
+import { createQueryFilterClauses } from '../../../../../utils/build_query';
+import { UsersRequestOptions } from '../../../../../../common/search_strategy/security_solution/users/all';
+import {
+  SortUsersField,
+  UsersFields,
+} from '../../../../../../common/search_strategy/security_solution/users/common';
+import { assertUnreachable } from '../../../../../../common/utility_types';
+
+export const buildUsersQuery = ({
+  defaultIndex,
+  docValueFields,
+  filterQuery,
+  pagination: { querySize },
+  sort,
+  timerange: { from, to },
+}: UsersRequestOptions): ISearchRequestParams => {
+  const filter = [
+    ...createQueryFilterClauses(filterQuery),
+    {
+      range: {
+        '@timestamp': {
+          gte: from,
+          lte: to,
+          format: 'strict_date_optional_time',
+        },
+      },
+    },
+  ];
+
+  const agg = { user_count: { cardinality: { field: 'user.name' } } };
```

Comment:
why is this aggregation stored in a const, but the other `user_data` agg is hardcoded?